client/profile-wizard/header.js

```diff
@@ -2,11 +2,58 @@
  * External dependencies
  */
 import { Component } from '@wordpress/element';
-import { filter } from 'lodash';
+import { filter, isEqual } from 'lodash';
 import { Stepper } from '@woocommerce/components';
 import { updateQueryString } from '@woocommerce/navigation';
 
+/**
+ * Internal dependencies
+ */
+import UnsavedChangesModal from './unsaved-changes-modal';
+
 export default class ProfileWizardHeader extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			showUnsavedChangesModal: false,
+		};
+		this.lastClickedStepKey = null;
+	}
+
+	shouldWarnForUnsavedChanges( step ) {
+		if ( typeof this.props.stepValueChanges[ step ] !== 'undefined' ) {
+			return ! isEqual(
```

Comment:
There's a small maybe-bug here that results from the lack of canonicalization before comparison here (i.e the order of the objects in an array matter when doing isEqual).

This can be observed in this recording: https://app.replay.io/recording/setup-wizard-woocommerce-woocommerce-admin-woocommerce--2eb22ed1-47bd-432f-abff-50573e883515

Not sure its worth fixing because it'll involve going down the object tree and sorting all the arrays recursively. The impact to UX is quite small IMO, but could leave some users confused when they see the warning and then try restoring all the selection to avoid the warning

Reply:
Thank you @rjchow That's a good catch. I'll work on fixing it!

Reply:
Updated in https://github.com/woocommerce/woocommerce-admin/pull/8278/commits/6a79a6b6e2ede6643e71cf7c28fa9e89f193e17f and https://github.com/woocommerce/woocommerce-admin/pull/8278/commits/908407df0c93f33ae404bae44e9fd35ba856f5f6